service/history/mutableStateBuilder.go

```diff
@@ -124,9 +127,11 @@ type (
 		updateBufferedEvents []*historypb.HistoryEvent // buffered history events that needs to be persisted
 		clearBufferedEvents  bool                      // delete buffered events from persistence
 
-		executionInfo    *persistence.WorkflowExecutionInfo // Workflow mutable state info.
-		versionHistories *persistence.VersionHistories
-		hBuilder         *historyBuilder
+		executionInfo  *persistenceblobs.WorkflowExecutionInfo // Workflow mutable state info.
+		executionState *persistenceblobs.WorkflowExecutionState
+		nextEventID    int64
```

Comment:
why the next event ID is added here?

Reply:
It was part of old `WorkflowExecutionInfo` but it is stored as separate field in db, therefore it is not in proto struct.